cvat/apps/iam/serializers.py

```diff
@@ -38,3 +39,26 @@ def get_email_options(self):
         return {
             'domain_override': domain
         }
+
+class LoginSerializerEx(LoginSerializer):
+    def get_auth_user_using_allauth(self, username, email, password):
+        from allauth.account import app_settings
+
+        # Authentication through email
+        if app_settings.AUTHENTICATION_METHOD == app_settings.AuthenticationMethod.EMAIL:
+            return self._validate_email(email, password)
+
+        # Authentication through username
+        if app_settings.AUTHENTICATION_METHOD == app_settings.AuthenticationMethod.USERNAME:
+            return self._validate_username(username, password)
+
+        # Authentication through either username or email
+        if email:
+            try:
+                username = UserModel.objects.get(email__iexact=email).get_username()
+            except UserModel.MultipleObjectsReturned:
+                raise ValidationError('Unable to login with provided credentials')
+            except UserModel.DoesNotExist:
+                pass
```

Comment:
I guess this approach will be better(e.g [`USER_MODEL_EMAIL_FIELD`](https://github.com/pennersr/django-allauth/blob/4d61aded877c9e4644564bd2cfc8305ad37cd966/allauth/account/app_settings.py#L282) can be set to different value (default `email`)). What do you think?
```
from allauth.account.utils import filter_users_by_email
users = filter_users_by_email(email)
if not users or len(users) > 1:
	raise ValidationError('Unable to login with provided credentials')
```